src/api/app/controllers/attribute_controller.rb

```diff
@@ -87,54 +87,52 @@ def show_attribute_definition
     status: 404, errorcode: 'unknown_attribute'
   end
 
-  # /attribute/:namespace/:name/_meta
-  def attribute_definition
+  def delete_attribute_definition
     # permission check via User model
     return unless extract_user
 
-    if request.post? || request.put?
-      logger.debug '--- updating attribute type definitions ---'
-
-      xml_element = Xmlhash.parse(request.raw_post)
+    if at = @ans.attrib_types.where('name = ?', name).first
+      authorize at, :destroy?
+      at.destroy
+    end
 
-      unless xml_element && xml_element['name'] == name && xml_element['namespace'] == namespace
-        render_error status: 400, errorcode: 'illegal_request',
-          message: "Illegal request: PUT/POST #{request.path}: path does not match content"
-        return
-      end
+    render_ok
+  end
 
-      entry = @ans.attrib_types.where('name = ?', name).first
+  # /attribute/:namespace/:name/_meta
+  def update_attribute_definition
+    # permission check via User model
+    return unless extract_user
 
-      if entry
-        authorize entry, :update?
+    logger.debug '--- updating attribute type definitions ---'
 
-        db = AttribType.find(entry.id) # get a writable object
-        logger.debug '* updating existing attribute definitions'
-        db.update_from_xml(xml_element)
-      else
-        entry = AttribType.new(name: name, attrib_namespace: @ans)
-        authorize entry, :create?
+    xml_element = Xmlhash.parse(request.raw_post)
 
-        logger.debug '* create new attribute definition'
-        entry.update_from_xml(xml_element)
-      end
+    unless xml_element && xml_element['name'] == name && xml_element['namespace'] == namespace
+      render_error status: 400, errorcode: 'illegal_request',
+        message: "Illegal request: PUT/POST #{request.path}: path does not match content"
+      return
+    end
 
-      logger.debug '--- finished updating attribute namespace definitions ---'
-      #--- end update attribute namespace definitions ---#
-      render_ok
-    elsif request.delete?
-      at = @ans.attrib_types.where('name = ?', name).first
+    entry = @ans.attrib_types.where('name = ?', name).first
```

Comment:
you can write it like `@ans.attrib_types.find_by(name: name)` and this code is repeated... :disappointed:

Reply:
I know - I just moved lines